provider/pkg/resources/patching.go

```diff
@@ -0,0 +1,80 @@
+package resources
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/mattbaird/jsonpatch"
+	"github.com/pulumi/pulumi-aws-native/provider/pkg/metadata"
+	"github.com/pulumi/pulumi-aws-native/provider/pkg/naming"
+	"github.com/pulumi/pulumi/pkg/v3/codegen"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/wI2L/jsondiff"
+)
+
+func CalcPatch(oldInputs resource.PropertyMap, newInputs resource.PropertyMap, spec metadata.CloudAPIResource, types map[string]metadata.CloudAPIType) ([]jsonpatch.JsonPatchOperation, error) {
+	diff := oldInputs.Diff(newInputs)
+
+	// Write-only properties can't even be read internally within the CloudControl service so they must be included in
+	// patch requests as adds to ensure the updated model validates.
+	// If a property is both write-only and create-only, we should not include it in the patch request.
```

Comment:
nit. I'd like to spell it out more: "However, If a property is both write-only and create-only, we should not include it in the patch request, because create-only properties cannot be updated."